map/src/map/layers/PoiLayer.js

```diff
@@ -157,51 +157,58 @@ export default function PoiLayer() {
         ctx.setUpdateInfoBlock(true);
     }
 
-    function createPoiLayer(poiList = []) {
-        const layers = poiList.map((poi) => {
-            const coord = poi.geometry.coordinates;
-            return new L.Marker(new L.LatLng(coord[1], coord[0]), {
-                title: poi.properties.name,
-                icon: getPoiIcon(poi),
-                type: poi.properties.type,
-                subType: poi.properties.subType,
-                iconKeyName: poi.properties.iconKeyName,
-                typeOsmTag: poi.properties.typeOsmTag,
-                typeOsmValue: poi.properties.typeOsmValue,
-                iconName: poi.properties.iconName,
-                operator: poi.properties.operator,
-                website: poi.properties.website,
-                wikipedia: poi.properties.wikipedia,
-                opening_hours: poi.properties.opening_hours,
-                email: poi.properties.email,
-                phone: poi.properties.phone,
-                facebook: poi.properties.facebook,
-                instagram: poi.properties.instagram,
-                osmUrl: poi.properties.osmUrl,
-            });
-        });
+    async function createPoiLayer(poiList = []) {
+        const layers = await Promise.all(
+            poiList.map(async (poi) => {
+                const icon = await getPoiIcon(poi);
+                const coord = poi.geometry.coordinates;
+                return new L.Marker(new L.LatLng(coord[1], coord[0]), {
+                    title: poi.properties.name,
+                    icon: icon,
+                    type: poi.properties.type,
+                    subType: poi.properties.subType,
+                    iconKeyName: poi.properties.iconKeyName,
+                    typeOsmTag: poi.properties.typeOsmTag,
+                    typeOsmValue: poi.properties.typeOsmValue,
+                    iconName: poi.properties.iconName,
+                    operator: poi.properties.operator,
+                    website: poi.properties.website,
+                    wikipedia: poi.properties.wikipedia,
+                    opening_hours: poi.properties.opening_hours,
+                    email: poi.properties.email,
+                    phone: poi.properties.phone,
+                    facebook: poi.properties.facebook,
+                    instagram: poi.properties.instagram,
+                    osmUrl: poi.properties.osmUrl,
+                });
+            })
+        );
 
         if (layers.length) {
-            return new L.FeatureGroup(layers);
+            return L.featureGroup(layers);
+        } else {
+            return L.featureGroup(); // return an empty layer group if there are no layers
         }
     }
 
-    function getPoiIcon(poi) {
-        const svg = MarkerOptions.getBackground(PoiManager.DEFAULT_POI_COLOR, PoiManager.DEFAULT_SHAPE_COLOR);
+    async function getPoiIcon(poi) {
```

Comment:
Please avoid redunant fetch() that now leads to 100500 http-queries when POI are displayed on the map.